### ISSUE TITLE
Seems something is broken.. tried clearing and reimporting... 

### ISSUE THREAD

**Issue by cantoute** [johngodley/redirection#1972]
```
Plugin: 4.4
WordPress: 5.2.3 (single)
PHP: 7.2.22-1+0~20190902.26+debian9~1.gbpd64eb7
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36
JavaScript: https://www.welovetennis.fr/wp-content/plugins/redirection/redirection.js
REST API: https://www.welovetennis.fr/wp-json/

Error: Invalid redirect details (80)
Raw: {"code":"redirect","message":"Invalid redirect details","data":{"status":400,"error_code":80}}
```

**Comment by cantoute** [comment]
Crazy thing is when I clear everything and remove tables etc... the redirections reappear !!!

**Comment by cantoute** [comment]
hi,

I suggest this is reopen... I finally identified the problem.

And yes it is cache related, but the issue appears using superTWA plugin

the plugin is not maintained any more but works for most of the time really well... just that it caches requests to /wp-json/

in `/superpwa-sw.js`

```javascript
const filesToCache = [startPage, offlinePage];
// const neverCacheUrls = [/\/wp-admin/,/\/wp-login/,/preview=true/];
const neverCacheUrls = [/\/wp-admin/,/\/wp-login/,/preview=true/,/\/wp-json\//];
```
so it's not your fault, but I think this problem will reappear
I can imagine that there are cases in PWA context where we want to be able to cache those requests.
An easy way to avoid ever having this kind of problem (that can be disastrous + frustration etc...) and when you have a vision of the mess there is around wordpress, perhaps using 'POST' method would avoid any risks.

Yes it should be a GET ... but this is wp world and cache in there is a mess (dev are good at disabling cache by 20 ways for making dev easier) and is really tricky to cache what is needed and not what shouldn't (all use POST when they should use GET because dealing with cache in ajax not all understood how)

So to handle heavy load, common practice is to ignore any cache-control/expire headers before passing to varnish reverse proxy (on my setup anything having `/wp-.*` is ignored but superpwa was doing things out of varnish cache scope)

Hope this can help... Your extension has been helpful to me :)

**Comment by johngodley** [comment]
The REST API is listed under `neverCacheUrls`, so it shouldnt be cached?

**Comment by johngodley** [comment]
Or are you saying that is your modification to the plugin?

**Comment by cantoute** [comment]
yes, changing this to the `/superpwa-sw.js` fixes the problem... yes it does cache /wp-json/ requests

The plugin, all it does is add two url `/superpwa-sw.js` and `/superpwa-manifest.json`

I had never played with pwa before... so I didn't know what was going on under the hood (and had more urgent things to deal with)

so I did a simple `curl https://site.exemple/superpwa-sw.js > superpwa-sw.js` and same with json

I'm surprised you couldn't reproduce the issue back then... tri deleting a rule and you'll see it still there etc...

**Comment by johngodley** [comment]
Using POST doesn't really solve caching problems, and it's not something I want to change in the plugin to accommodate an issue that affects relatively few people. The REST API shouldn't be cached, and the PWA plugin does provide the filter `superpwa_sw_never_cache_urls` to add more URLs to ignore. Hopefully an update will include the REST API by default, but until then using `superpwa_sw_never_cache_urls` seems the way to go.

**Comment by cantoute** [comment]
As you like... but the superPWA doesn't offer this option... I just short circuited it by putting this file at site root.

So here I'm in wp world... The theme guy use POST for ajax request on front end that I should cache but can't, and I have an admin plugin that uses GET that I have to short circuit... what a marvelous world you guys live in!

I'm just glad this is my last ever wordpress project.
I did this for a friend and don't have time to write this in Angular... but I learned my lesson!
